src/Uno.UI/UI/Xaml/Controls/ContentControl/ContentControl.skia.cs

```diff
@@ -25,6 +25,17 @@ public partial class ContentControl
 			RemoveChild(ContentTemplateRoot);
 		}
 
-		protected override Size MeasureOverride(Size availableSize) => base.MeasureOverride(availableSize);
+		protected override Size MeasureOverride(Size availableSize)
+		{
+			// This is the same as Control.MeasureOverride, except that we use Content if FindFirstChild is null.
+			// The fact that FindFirstChild is null is actually a BUG!
+			// The way this works in WinUI is:
+			// 1. FrameworkElement.MeasureCore calls 'InvokeApplyTemplate'
+			// 2. ContentControl overrides ApplyTemplate to do a bunch of stuff, including a call to CreateDefaultVisuals.
+			// 3. CreateDefaultVisuals adds the Content as Child if it's a UIElement.
+			// Tracking issue: https://github.com/unoplatform/uno/issues/190
+			var child = this.FindFirstChild() ?? (Content as UIElement);
+			return child != null ? MeasureElement(child, availableSize) : new Size(0, 0);
```

Comment:
Looks like this is causing another issue:

https://dev.azure.com/uno-platform/Uno%20Platform/_TestManagement/Runs?runId=2583081&_a=resultSummary&resultId=101519

Could be that we were testing for an invalid behavior, though:
https://github.com/unoplatform/uno/blob/7a65f71b34f7bdb5fb7648f2423ed529c342ec50/src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Given_FrameworkElement.cs#L266

Reply:
@jeromelaban This test also fails in WinUI. However, there is another NaN-measuring test that doesn't fail in WinUI. So I still need to figure that out 👀

Reply:
@jeromelaban The exception on WinUI is a weird "COMException". It happens when `Measure` is called from C# with NaN. However, if `Measure` is called with `NaN` somewhere in the C++ code, it is allowed.

I think it should be acceptable to modify this test and expect no exception be thrown?

To see this behavior on WinUI, use this code

```csharp
            var grid = new Grid();
            var SUT = new MyControl02();
            SUT.Content = new Grid();
            grid.Children.Add(SUT);
            SUT.BaseAvailableSize = new Size(double.NaN, double.NaN);
            grid.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
```

and

```csharp
    public partial class MyControl02 : ContentControl
    {
        public List<Size> MeasureOverrides { get; } = new List<Size>();

        public Size? BaseAvailableSize;

        protected override Size MeasureOverride(Size availableSize)
        {
            MeasureOverrides.Add(availableSize);
            //(Content as UIElement).Measure(BaseAvailableSize ?? availableSize);
            //return (Content as UIElement).DesiredSize;
            return base.MeasureOverride(BaseAvailableSize ?? availableSize);
        }
    }
```

The above code will work without any crashes/exceptions.
However, `base.MeasureOverride` will end up calling `ContentControl.MeasureOverride` which in turn will call `Measure` with NaN.

Now, if you remove the `base.MeasureOverride` line and uncomment the two lines above it, it will crash.

![image](https://github.com/unoplatform/uno/assets/31348972/03570c41-61f8-4184-ac9d-1f96996d8b01)

Reply:
This was likely present to align with some original behavior. Did you take a look at why this test was originally added?

Otherwise, if adjusting the test matches more with what WinUI does, I'm good with it.

Reply:
@jeromelaban https://github.com/unoplatform/private/issues/93